nibabel/testing/__init__.py

```diff
@@ -79,12 +103,44 @@ def __enter__(self):
                 mod_reg = mod.__warningregistry__
                 self._warnreg_copies[mod] = mod_reg.copy()
                 mod_reg.clear()
-        return super(catch_warn_reset, self).__enter__()
+        return super(clear_and_catch_warnings, self).__enter__()
 
     def __exit__(self, *exc_info):
-        super(catch_warn_reset, self).__exit__(*exc_info)
+        super(clear_and_catch_warnings, self).__exit__(*exc_info)
         for mod in self.modules:
             if hasattr(mod, '__warningregistry__'):
                 mod.__warningregistry__.clear()
             if mod in self._warnreg_copies:
                 mod.__warningregistry__.update(self._warnreg_copies[mod])
+
+
+class error_warnings(clear_and_catch_warnings):
+    """ Context manager to check for warnings as errors.  Usually used with
+    ``assert_raises`` in the with block
+
+    Examples
+    --------
+    >>> with error_warnings():
+    ...     try:
+    ...         warnings.warn('Message', UserWarning)
+    ...     except UserWarning:
+    ...         print('I consider myself warned')
+    I consider myself warned
+    """
+    filter = 'error'
+
+    def __enter__(self):
+        mgr = super(error_warnings, self).__enter__()
+        warnings.simplefilter(self.filter)
+        return mgr
+
+
+class suppress_warnings(error_warnings):
+    filter = 'ignore'
+
+
+class catch_warn_reset(clear_and_catch_warnings):
```

Comment:
Removed docstring, and all functions take `*args`, `**kwargs` for all deprecated classes.